pkg/permissionclaim/permissionclaim_labeler.go

```diff
@@ -136,3 +140,105 @@ func (l *Labeler) LabelsFor(ctx context.Context, cluster logicalcluster.Name, gr
 
 	return labels, nil
 }
+
+// ObjectPermitted determines if any PermissionClaims apply to a given object's GroupResource, namespace, or name.
+// If no APIBinding or PermissionClaim exists for the GroupResource, the object is permitted.
+func (l *Labeler) ObjectPermitted(ctx context.Context, cluster logicalcluster.Name, groupResource schema.GroupResource, resourceName, resourceNamespace string) (bool, error) {
+	bindings, err := l.listAPIBindingsAcceptingClaimedGroupResource(cluster, groupResource)
+	if err != nil {
+		return true, fmt.Errorf("error listing APIBindings in %q accepting claimed group resource %q: %w", cluster, groupResource, err)
+	}
+
+	logger := klog.FromContext(ctx)
+
+	for _, binding := range bindings {
+		logger := logging.WithObject(logger, binding).WithValues(
+			"namespace", resourceNamespace, "name", resourceName,
+		)
+
+		path := logicalcluster.NewPath(binding.Spec.Reference.Export.Path)
+		if path.Empty() {
+			path = logicalcluster.From(binding).Path()
+		}
+
+		for _, claim := range binding.Spec.PermissionClaims {
+			// There is no PermissionClaim for this object, no need to compute selection
+			if claim.State != apisv1alpha1.ClaimAccepted || claim.Group != groupResource.Group || claim.Resource != groupResource.Resource {
+				continue
+			}
+			// The permission claim is relevant for this object, but the object does not match the criteria
+			if !Selects(claim, resourceName, resourceNamespace) {
+				// return if any permission claim on any binding do not allow the object.
+				logger.Info("not permitted by PermissionClaims")
+				return false, fmt.Errorf("operation not permitted")
```

Comment:
why do we need this error? This is a predicate. It should not judge about what the result means.